src/parser/Parser.ts

```diff
@@ -3543,3 +3559,10 @@ export function getBscTypeFromExpression(expression: Expression, functionExpress
     //fallback to dynamic
     return new DynamicType();
 }
+
+/**
+ * A function that does nothing
+ */
+function noop(...args) {
```

Comment:
What is this for? lol

Reply:
It's used in relation to this part of my description: `To handle their range calculations, we call the get range getter after they are constructed, which finalizes the range property.` 

Here's one of the places it's used: https://github.com/rokucommunity/brighterscript/pull/526/files#diff-0d27a1b5098cc3c9f464ebe55376e540e5ffd0779dd682227eb284abf03e6478R680

Which results in things like this getting cached at parse-time: https://github.com/rokucommunity/brighterscript/pull/526/files#diff-8f54ec4bf9e1be0f718a7b8712e8ba4f9df4f443432f19e34e1b60aed9551874R159

Reply:
hummmm, could we not do this by having a function on the expression it self? like `expression.casheRange()` or something? It's kind of a magic thing thats happening when done in this was currently.

Reply:
Yeah, fair enough. I'll change it.

Reply:
Changed. Please review again.